src/hid_linux.c

```diff
@@ -165,6 +173,26 @@ copy_info(fido_dev_info_t *di, struct udev *udev,
 		di->manufacturer = strdup("");
 	if ((di->product = get_usb_attr(dev, "product")) == NULL)
 		di->product = strdup("");
+	if ( di->manufacturer != NULL && *di->manufacturer == '\0' && 
+		 di->product != NULL && *di->product == '\0' && 
+		 hid_name != NULL ) {
+		char* first_space = strchr(hid_name, ' ');
```

Comment:
While I can understand the intention here, I'd probably be more in favor of not manipulating the string in any way and punting that potential work to the application.  This would simplify handling here: e.g. if `di->manufacturer` and `di->product` are both unset, set `di->product` to `hid_name`.

Reply:
I'm fine either way. Technically, just splitting at the first space is not guaranteed to be correct, as the manufacturer string might include spaces itself, e.g. `Yubico AB`. In practice, most companies try to make it a single word and leave out the ` AB` part in this string. I'll follow your guidance/stated preference.